File: FlightsApp/Backend/Flights/DTOs/NewFlightDTO.cs
namespace Flights.DTOs
{
    public class NewFlightDTO
    {

        public string departurePoint { get; set; }
        public string arrivalPoint { get; set; }
        public string departureTime { get; set; }
        public int duration { get; set; }
        public  float ticketPrice { get; set; }
        public int numberOfPassengers { get; set; }
        public int remainingTickets { get; set; }

        public NewFlightDTO()
        {
        }

        public NewFlightDTO(string departurePoint, string arrivalPoint, string departureTime, int duration, float ticketPrice, int numberOfPassengers, int remainingTickets)
        {
            this.departurePoint = departurePoint;
           this.arrivalPoint = arrivalPoint;
            this.departureTime = departureTime;
            this.duration = duration;
           this.ticketPrice = ticketPrice;
            this.numberOfPassengers = numberOfPassengers;
            this.remainingTickets = remainingTickets;
        }
    }
}

File: FlightsApp/Frontend/Flights/src/app/model/newFlightDto.ts
export class NewFlightDto {
    
    departurePoint?: string;
    arrivalPoint?:  string;
    departureTime?: string;
    duration: number=0;
    ticketPrice: number=0;
    numberOfPassengers: number=0;
    remainingTickets: number=0;

    public constructor(obj?: any){
        if(obj){
            this.departurePoint = obj.DeparturePoint;
            this.arrivalPoint = obj.ArrivalPoint;
            this.departureTime = obj.DepartureTime;
            this.duration = obj.Duration;
            this.ticketPrice = obj.TicketPrice;
            this.numberOfPassengers = obj.NumberOfPassengers;
            this.remainingTickets = obj.RemainingTickets;
        }
    }

}